src/Raven.Server/Documents/DatabaseRequestHandler.cs

```diff
@@ -40,10 +40,6 @@ public override void Init(RequestHandlerContext context)
 
         public Task CheckForChanges(RequestHandlerContext context)
         {
-            var topologyEtag = GetLongFromHeaders(Constants.Headers.TopologyEtag);
-            if (topologyEtag.HasValue && Database.HasTopologyChanged(topologyEtag.Value))
```

Comment:
you are checking here database topology and in server you are checking cluster
also no you are subscribing twice to the OnStarting event, would prefer to handle that once

Reply:
My bad. Also changed back so it will inherit from RequestHandler